sphinx_immaterial/apidoc_formatting.py

```diff
@@ -11,6 +15,25 @@
 _ = sphinx.locale._
 
 
+if sys.version_info >= (3, 8):
+
+    class ObjectDescriptionOptions(typing.TypedDict):
+        domain: str
+        object_type: str
+        wrap_signatures_with_css: bool
+        wrap_signatures_column_limit: int
+        include_in_toc: bool
+        generate_synopses: Literal["first_paragraph", "first_sentence", None]
+        include_object_type_in_xref_tooltip: bool
+        include_object_description_fields_in_toc: bool
+        toc_icon_text: Optional[str]
+        toc_icon_class: Optional[str]
+
+
+else:
+    ObjectDescriptionOptions = dict
+
+
 if TYPE_CHECKING:
```

Comment:
We're not using mypy for type checking in CI. This feels specific to using mypy after looking into it. Pylint doesn't do type checking to the extent that mypy does - mypy is the org that introduced type checking in python and provided the mypy tool to make proper use of `typing` module.

Reply:
We might want to use mypy in the future, but for now I've removed this.

Reply:
Good idea for a separate PR. I tend to avoid mypy out of laziness, but it makes sense... I primarily use type checking to augment VSCode's intellisense tool (& "go to definition" option).